airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/DestinationTaskScopeProvider.kt

```diff
@@ -64,31 +71,42 @@ class DestinationTaskScopeProvider(config: DestinationConfiguration) :
                 .asCoroutineDispatcher()
         )
 
-    override suspend fun launch(task: ScopedTask) {
-        when (task) {
-            is InternalTask -> internalScope.scope.launch { execute(task) }
-            is ImplementorTask -> implementorScope.scope.launch { execute(task) }
+    override suspend fun launch(task: WrappedTask<ScopedTask>) {
+        when (task.innerTask) {
+            is InternalScope -> internalScope.scope.launch { execute(task, "internal") }
+            is ImplementorScope -> implementorScope.scope.launch { execute(task, "implementor") }
+            is ShutdownScope -> implementorScope.scope.launch { execute(task, "shutdown") }
         }
     }
 
-    private suspend fun execute(task: ScopedTask) {
-        log.info { "Launching task $task" }
+    private suspend fun execute(task: WrappedTask<ScopedTask>, scope: String) {
+        log.info { "Launching task $task in scope $scope" }
         val elapsed = measureTimeMillis { task.execute() }
         log.info { "Task $task completed in $elapsed ms" }
     }
 
-    override suspend fun close() = supervisorScope {
+    override suspend fun close() {
         log.info { "Closing task scopes" }
-        internalScope.job.cancel()
         // Under normal operation, all tasks should be complete
         // (except things like force flush, which loop). So
         // - it's safe to force cancel the internal tasks
-        // - implementor scope should join immediately unless we're
-        //   failing, in which case we want to give them a chance to
-        //   fail gracefully
+        // - implementor scope should join immediately
+        internalScope.job.cancel()
+        log.info { "Cancelled internal tasks, waiting for implementor tasks to complete" }
+        implementorScope.job.join()
```

Comment:
doublechecking: when we get here, the only queued/in-flight implementor tasks are things that happen between PERSISTED and COMPLETE?

Reply:
This is defensive. Under a successful close, we should never get here, because `TeardownTask` will wait until after this returns to mark the sync successful, and it should have not run until everything was done. My main reason for having this is so that

1) if I screw that up we end up with the job hanging forever in a test, or
2) if we fail to catch an exception elsewhere, it will bubble up here (again, before we've marked the sync successful)